src/routes.jsx

```diff
@@ -18,12 +19,17 @@ const router = createBrowserRouter([
         element: <ProtectedRoute />,
         children: [
           {
-            path: "/:coachId/generate-referral-link",
-            element: <GenerateReferralLink />
-          },
-          {
-            path: "/:coachId/create-exercise",
-            element: <CreateExerciseForm />
+            element: <CoachProtectedRoute />,
```

Comment:
How we have `<ProtectedRoute />` that authenticates the user, and the recently defined `<CoachProtectedRoute />` to authorize the `Choach` user type to access the pages included in the `children` attribute of the middleware component.